pkg/deploy/api/validation/validation.go

```diff
@@ -39,7 +39,18 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// rather than handleEmptyImageReferences returning an array of changed containers, PR review deemed it more
+		// efficient to build a list of the original container to image mappings, and then just restore all of them (and hence,
+		// even the ones not modified)
+		originalContainerImages := make(map[string]string)
+		for _, container := range spec.Template.Spec.Containers {
+			originalContainerImages[container.Name] = container.Image
+		}
+		handleEmptyImageReferences(spec.Template, spec.Triggers)
```

Comment:
a map has the potential to restore something other than the original image names. I'd still prefer helper funcs to get the image names as an array, and to set the image names from an array, using them like
```
originalNames := getContainerImageNames(spec.Template)
defer setContainerImageNames(spec.Template, originalNames)
```